dvc/objects/db/base.py

```diff
@@ -360,6 +396,9 @@ def gc(self, used, jobs=None):
             self.fs.remove(path_info)
             removed = True
 
+        if self.staging is not None:
+            self.staging.gc(set(), jobs=jobs)
```

Comment:
Again, pretty weird that odb itself knows that it is for staging and skips gc here. Am I missing something?

Reply:
If the parent ODB has a staging child ODB, it gc's it.

The child ODB won't know itself that it is for staging (it will have a None `staging` value)